StereoKitC/tools/file_picker.cpp

```diff
@@ -435,30 +461,114 @@ void file_picker_update() {
 			if (fp_active == nullptr) ui_push_enabled(false);
 			if (ui_button("Open")) { snprintf(fp_filename, sizeof(fp_filename), "%s%c%s", fp_path.folder, platform_path_separator_c, fp_active); fp_call = true; fp_call_status = true; }
 			ui_sameline();
+			ui_push_enabled(true);
+			if (ui_button("ListMode")) {
+				fp_list_mode = !fp_list_mode;
+			}
+			ui_sameline();
+			if (fp_active == nullptr) ui_push_enabled(false);
 			ui_label(fp_active?fp_active:"None selected...");
 			if (fp_active == nullptr) ui_pop_enabled();
 		} break;
 		}
 
 		ui_hseparator();
 
+		sprite_t spr_folder = sprite_find(ui_default_id_spr_arrow_right);
+		static ui_btn_layout_ ui_folderimg_button_layout = ui_btn_layout_left;
 		// List the files
 		vec2 size = { .12f, line_height * 1.5f };
+		if (fp_list_mode) {
+			sprite_t spr_up = sprite_find(ui_default_id_spr_arrow_up);
+			static ui_btn_layout_ layout = ui_btn_layout_left;
+			if (fp_sortby == Name) {
+				if (ui_button_img_sz("Name", spr_up, layout, size)) {
+					fp_sortby = Name;
+					sort_order_changed = true;
+				}
+			}
+			else {
+				if (ui_button_sz("Name", size)) {
+					fp_sortby = Name;
+					sort_order_changed = true;
+				}
+			}
+			ui_sameline();
+			ui_text_sz(" ", text_align_x_center, text_fit_clip, size);
```

Comment:
Spacing like this can vary based on the font used! I'd prefer to see spacing done by calculating more specific sizes for elements, and obey the theme's rules for gutters. See `ui_layout_remaining` and `ui_get_settings`.